libs/langchain/langchain/document_loaders/batch_loader.py

```diff
@@ -0,0 +1,252 @@
+import asyncio
+import multiprocessing
+from concurrent.futures import ThreadPoolExecutor
+from functools import partial
+from typing import Any, Callable, Dict, Iterable, List, TypeVar
+
+from langchain.document_loaders.base import BaseLoader
+from langchain.schema.document import Document
+
+T = TypeVar("T")
+
+
+def make_iterator_list(
+    iterable: Iterable[T],
+    args_length: int,
+    show_progress: bool = False,
+) -> List[T]:
+    """Make an iterator from an iterable. If show_progress is True, will
+    return a tqdm iterator.
+
+    Args:
+        iterable (Iterable[T]): any iterable
+        args_length (int): length of the iterable
+        show_progress (bool, optional): If true, will return a tqdm iterator.
+            Defaults to False.
+
+    Raises:
+        ImportError: if tqdm is not installed and show_progress is True
+    """
+    if show_progress:
+        try:
+            from tqdm import tqdm
+        except ImportError as err:
+            raise ImportError(
+                "You must install tqdm to use show_progress=True."
+                "You can install tqdm with `pip install tqdm`."
+            ) from err
+        return list(tqdm(iterable, total=args_length))
+    return list(iterable)
+
+
+class BatchLoader(BaseLoader):
+    """
+    Meta loader use to load multiple data from a loader either in parallel
+    or sequentially.
+
+    Example:
+
+    Using a class that is instantiate with __init__
+    ```
+    file_path_list : List[str] = [...]
+    loader_files = BatchLoader(
+        TextLoader,
+        {
+            "file_path": file_path_list
+        },
+    )
+    ```
+    Using a static method to initialize the loader:
+    ```
+    youtube_video_urls : List[str] = [...]
+    loader_youtube = BatchLoader(
+        YoutubeLoader.from_url,
+        {
+            "url": youtube_video_urls
+        },
+    )
+    ```
+
+    Then you can load your data:
+    ```
+    documents = loader_files.load(method="thread", max_workers=4)
+    documents = loader_youtube.load(method="process", max_workers=4)
+    documents = loader_youtube.load(method="sequential")
+    documents = loader_youtube.load(method="async")
+    ```
+    """
+
+    def __init__(
+        self,
+        loader_initializer_callable: Callable[..., BaseLoader],
+        init_loader_args: Dict[str, List[Any]],
```

Comment:
Could you use `*` before any named arguments?

Reply:
Done in last commit